src/components/checkboxes/template.test.js

```diff
@@ -0,0 +1,241 @@
+        'data-attribute': 'value',
+        'data-second-attribute': 'second-value'
+      }
+    })
+
+    const $component = $('.govuk-c-checkboxes')
+
+    expect($component.attr('data-attribute')).toEqual('value')
+    expect($component.attr('data-second-attribute')).toEqual('second-value')
+  })
+
+  describe('items', () => {
+    it('render a matching label and input using name by default', () => {
+      const $ = render('checkboxes', {
+        name: 'example-name',
+        items: [
+          {
+            value: '1',
+            text: 'Option 1'
+          },
+          {
+            value: '2',
+            text: 'Option 2'
+          }
+        ]
+      })
+
+      const $component = $('.govuk-c-checkboxes')
+
+      const $firstInput = $component.find('.govuk-c-checkboxes__item:first-child input')
+      const $firstLabel = $component.find('.govuk-c-checkboxes__item:first-child label')
+      expect($firstInput.attr('id')).toEqual('example-name-1')
+      expect($firstLabel.attr('for')).toEqual('example-name-1')
+
+      const $lastInput = $component.find('.govuk-c-checkboxes__item:last-child input')
+      const $lastLabel = $component.find('.govuk-c-checkboxes__item:last-child label')
+      expect($lastInput.attr('id')).toEqual('example-name-2')
+      expect($lastLabel.attr('for')).toEqual('example-name-2')
+    })
+
+    it('render a matching label and input using custom idPrefix', () => {
+      const $ = render('checkboxes', {
+        idPrefix: 'custom',
+        name: 'example-name',
+        items: [
+          {
+            value: '1',
+            text: 'Option 1'
+          },
+          {
+            value: '2',
+            text: 'Option 2'
+          }
+        ]
+      })
+
+      const $component = $('.govuk-c-checkboxes')
+
+      const $firstInput = $component.find('.govuk-c-checkboxes__item:first-child input')
+      const $firstLabel = $component.find('.govuk-c-checkboxes__item:first-child label')
+      expect($firstInput.attr('id')).toEqual('custom-1')
+      expect($firstLabel.attr('for')).toEqual('custom-1')
+
+      const $lastInput = $component.find('.govuk-c-checkboxes__item:last-child input')
+      const $lastLabel = $component.find('.govuk-c-checkboxes__item:last-child label')
+      expect($lastInput.attr('id')).toEqual('custom-2')
+      expect($lastLabel.attr('for')).toEqual('custom-2')
+    })
+
+    it('render disabled', () => {
+      const $ = render('checkboxes', {
+        name: 'example-name',
+        items: [
+          {
+            value: '1',
+            text: 'Option 1',
+            disabled: true
+          },
+          {
+            value: '2',
+            text: 'Option 2'
+          }
+        ]
+      })
+
+      const $component = $('.govuk-c-checkboxes')
+
+      const $firstInput = $component.find('.govuk-c-checkboxes__item:first-child input')
+      expect($firstInput.attr('disabled')).toEqual('disabled')
+    })
+
+    it('render checked', () => {
```

Comment:
Could we maybe increase this example to have three inputs and check that two inputs can be checked at one time?

Reply:
Yep. Done.